src/passport-saml/saml.ts

```diff
@@ -107,100 +107,111 @@ async function promiseWithNameID(nameid: Node): Promise<NameID> {
-     * - maximum:  Assertion context must be no stronger than a context in the list
-     * - better:  Assertion context must be stronger than all contexts in the list
-     */
-    if (
-      !options.RACComparison ||
-      ["exact", "minimum", "maximum", "better"].indexOf(options.RACComparison) === -1
-    ) {
-      options.RACComparison = "exact";
-    }
+    const options = {
+      ...this.ctorOptions,
+      passive: this.ctorOptions.passive ?? false,
+      disableRequestedAuthnContext: this.ctorOptions.disableRequestedAuthnContext ?? false,
+      additionalParams: this.ctorOptions.additionalParams ?? {},
+      additionalAuthorizeParams: this.ctorOptions.additionalAuthorizeParams ?? {},
+      additionalLogoutParams: this.ctorOptions.additionalLogoutParams ?? {},
+      forceAuthn: this.ctorOptions.forceAuthn ?? false,
+      skipRequestCompression: this.ctorOptions.skipRequestCompression ?? false,
+      disableRequestAcsUrl: this.ctorOptions.disableRequestAcsUrl ?? false,
+      acceptedClockSkewMs: this.ctorOptions.acceptedClockSkewMs ?? 0,
+      path: this.ctorOptions.path ?? "/saml/consume",
+      host: this.ctorOptions.host ?? "localhost",
+      issuer: this.ctorOptions.issuer ?? "onelogin_saml",
+      identifierFormat:
+        this.ctorOptions.identifierFormat === undefined
+          ? "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress"
+          : this.ctorOptions.identifierFormat,
+      authnContext: this.ctorOptions.authnContext ?? [
+        "urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport",
+      ],
+      validateInResponseTo: this.ctorOptions.validateInResponseTo ?? false,
+      cert: assertRequired(this.ctorOptions.cert, "cert is required"),
+      requestIdExpirationPeriodMs: this.ctorOptions.requestIdExpirationPeriodMs ?? 28800000, // 8 hours
+      cacheProvider:
+        this.ctorOptions.cacheProvider ??
+        new InMemoryCacheProvider({
+          keyExpirationPeriodMs: this.ctorOptions.requestIdExpirationPeriodMs,
+        }),
+      logoutUrl: this.ctorOptions.logoutUrl ?? this.ctorOptions.entryPoint ?? "", // Default to Entry Point
+      signatureAlgorithm: this.ctorOptions.signatureAlgorithm ?? "sha1", // sha1, sha256, or sha512
+      authnRequestBinding: this.ctorOptions.authnRequestBinding ?? "HTTP-Redirect",
+
+      RacComparison: (() => {
+        /**
+         * List of possible values:
+         * - exact : Assertion context must exactly match a context in the list
+         * - minimum:  Assertion context must be at least as strong as a context in the list
+         * - maximum:  Assertion context must be no stronger than a context in the list
+         * - better:  Assertion context must be stronger than all contexts in the list
+         */
+        this.ctorOptions.RacComparison = this.ctorOptions.RacComparison ?? "exact";
+        if (
+          ["exact", "minimum", "maximum", "better"].indexOf(this.ctorOptions.RacComparison) === -1
```

Comment:
`["exact", "minimum", "maximum", "better"].includes(this.ctorOptions.RacComparison)` would be more readable in my opinion. But maybe you choose to change as little as possible which would be fine also.